extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+    Downloads the installer packages from the url to the local folder. If the installer is already downloaded, then the
+    download is aborted unless the force flag is set to true.
+
+    .OUTPUTS
+    [string]: Path to the downloaded install package.
+    #>
+	Param(
+		# Full name of the installer package file which will be saved
+		[Parameter(Mandatory)]
+		[string] $packageName,
+		# Url from where to download the installer package
+		[Parameter(Mandatory)]
+		[string] $url,
+		# Download folder where the installer package will be downloaded
+		[Parameter(Mandatory)]
+		[string] $downloadFolder,
+		# Flag which defines if the installer should be downloaded even if it already exists
+		[Parameter()]
+		[switch] $force = $false
+	)
+
+	$packagePath = Join-Path $downloadFolder $packageName
+	$shouldDownload = $true
+
+	# If the package already exists then the download flag respects the force flag
+	if (Test-Path -path $packagePath) {
+		$shouldDownload = $force
+	}
+
+	if ($shouldDownload) {
+		# Check for internet connection before going forward
+		if (-not (Test-InternetConnectivity)) {
+			return
+		}
+
+		if (Test-Path -Path $packagePath) {
+			# Package already exists, force download
+			Remove-Item -Path $packagePath | Out-Null
+			Write-OutputAndLog "Removed cached package from $downloadFolder."
+		}
+
+		if (-not (Test-Path -Path $downloadFolder)) {
+			# If folder does not exists then create
+			New-Item $downloadFolder -ItemType Directory | Out-Null
+			Write-OutputAndLog "Created folder $downloadFolder."
+		}
+
+		Write-OutputAndLog "Downloading from url $url..."
+		Write-OutputAndLog "Downloading $packageName package..."
+		Invoke-WebRequest -Uri $url -OutFile $packagePath
+		Write-OutputAndLog "Package $packageName downloaded at $downloadFolder."
+		Write-OutputAndLog ('Downloaded {0} bytes.' -f (Get-Item $packagePath).length)
+	}
+	else {
+		# Display message that file already exists
+		Write-OutputAndLog "Package $packageName found at $downloadFolder."
+	}
+
+	return $packagePath
+}
+
+Function Test-InternetConnectivity {
+	<#
+    .DESCRIPTION
+    Checks the internet connection by pinging known sites and stops execution if internet connection is
+    not available.
+
+    .OUTPUTS
+    $true if the connection is available, otherwise $false
+    #>
+	Param (
+		[Parameter(Mandatory = $false)]
+		# Microsoft NCSI url for testing internet connectivity
+		[string]$Url = "http://www.msftncsi.com/ncsi.txt"
+	)
+
+	try {
+		Write-OutputAndLog "Checking for internet connection..."
+		$response = Invoke-WebRequest -Uri $Url -UseBasicParsing -ErrorAction SilentlyContinue
+		if ($null -ne $response) {
+			Write-OutputAndLog "Internet is connected."
+			return $true
+		}
+		Write-ErrorAndLog "Internet is not available on the machine and its a requirement for the execution of the script."
+		return $false
+	}
+	catch {
+		Write-ErrorAndLog "Internet is not available on the machine and its a requirement for the execution of the script."
```

Comment:
Prerequisites are not met. Internet connectivity is required on the machine.

Reply:
Updated